lib/tidewave/railtie.rb

```diff
@@ -22,6 +23,27 @@ class Railtie < Rails::Railtie
         Tidewave::Middleware,
         app.config.tidewave
       )
+
+      app.config.after_initialize do
+        # If the user configured CSP, we need to alter it in dev
+        # to allow TC to run browser_eval.
+        app.config.content_security_policy.try do |content_security_policy|
+          content_security_policy.directives["script-src"].try do |script_src|
+            script_src << "'unsafe-eval'" unless script_src.include?("'unsafe-eval'")
+          end
+        end
+      end
+    end
+
+    initializer "tidewave.intercept_exceptions" do |app|
+      # We intercept exceptions from DebugExceptions, format the
+      # information as text and inject into the exception page html.
+
+      ActionDispatch::DebugExceptions.register_interceptor do |request, exception|
+        request.set_header("tidewave.exception", exception)
+      end
+
+      app.middleware.insert_before(ActionDispatch::DebugExceptions, Tidewave::ExceptionsMiddleware)
```

Comment:
I am assuming we cannot reuse the existing middleware because they are added to different parts of the stack, right?